Core: Vite integration

50rayn opened the pull request:
In this PR done a couple of things: 

- upgrade existing Vite modules to 6
- in @unovis/ts package was replaced rollup with Vite

During integration, I met some misunderstandings, like:

#### Why in `.npmrc` do we have `legacy-peer-deps=true`? 
Because of this flag, I could not install the latest version of `vite-plugin-dts`.

#### In `@unovis/ts` in tsconfig, do we have a lot of `paths` configured? Why?
It makes more ambiguous the imports in the files. You may not understand it's imported from a file, module, and alias. I'd add just one, `"@/*": ["src/*"]`. It's more common for most users. 

I wanted to be sure that I could open some PRs for each of the mentions, but first of all, I need to be sure if it makes sense.

50rayn committed:
Vue | Solid: Upgrade to Vite 6

50rayn committed:
Core: Vite integration

rokotyan commented:
@50rayn Another great contribution, thanks!

The build looks good, but this dependencies should not be bundled:
<img width="391" alt="image" src="https://github.com/user-attachments/assets/3a35f18e-f1a7-4859-9e9e-249962f95eee">

> Why in .npmrc do we have legacy-peer-deps=true?

To be honest, I don't remember. You can try removing it and see what happens.

> In @unovis/ts in tsconfig, do we have a lot of paths configured? Why?

I agree with you. At the beginning of the project it was convenient this way but now it's not. We can change it to a single `"@/*"` alias.

50rayn commented:
Regarding both questions, I can do separate PRs. There, I'll check all the necessary changes.

About those dependencies, not problem, I'll take a look.

rokotyan commented:
@50rayn Thanks! We're planning to release Unovis 1.5 soon. Will get back to this right after

lee00678 commented:
Hi @50rayn, Thanks for the contribution! 

There seems to be an issue with dynamic import moving from rollup to vite. I released a beta version `[1.5.1-ql.2](https://www.npmjs.com/package/@unovis/ts/v/1.5.1-ql.2)` from this branch  and tested it in an angular project.  Some of the following path (https://github.com/f5/unovis/blob/main/packages/ts/src/components/leaflet-flow-map/renderer.ts#L1-L9) didn't get resolved correctly.

rokotyan commented:
Confirmed. There's a problem with imports that end with the `.js` extension, for example:
`import { Color } from 'three/src/math/Color.js'`. Once compiled they have the full local file system path:
```ts
import { Color as r } from "/Users/nikitarokotyan/unovis/node_modules/three/src/math/Color.js";
```

50rayn commented:
Thanks, @rokotyan  and @lee00678, for the feedback.  
I'll take a look.

rokotyan commented:
@50rayn I've tried to reconfigure the common js plugin a couple of times but without success. So now I'm out of ideas.

50rayn commented:
Hello everyone.
Glad to be back. I took a look at the issue. It was pretty challenging to find the issue with paths, and it was fixed (I hope so 😄). The only thing I could not play with was angular. The `gallery` script gives me errors (even in prod): 
> An unhandled exception occurred: Cannot find module 'webpack'

lee00678 commented:
> Hello everyone. Glad to be back. I took a look at the issue. It was pretty challenging to find the issue with paths, and it was fixed (I hope so 😄). The only thing I could not play with was angular. The `gallery` script gives me errors (even in prod):
> 
> > An unhandled exception occurred: Cannot find module 'webpack'

Thanks for the update. I will test it this week.

lee00678 commented:
> > Hello everyone. Glad to be back. I took a look at the issue. It was pretty challenging to find the issue with paths, and it was fixed (I hope so 😄). The only thing I could not play with was angular. The `gallery` script gives me errors (even in prod):
> > > An unhandled exception occurred: Cannot find module 'webpack'
> 
> Thanks for the update. I will test it this week.

@50rayn have you tried to do `npm run dev`? I'm getting errors after running a build. 
![Screenshot 2025-03-13 at 12 20 07 PM](https://github.com/user-attachments/assets/68633ffa-e04d-4a4a-b815-c35fdd228e53)

And if you look at the pipeline: https://github.com/f5/unovis/actions/runs/13773780019/job/38734324725?pr=495
Seems like it also fails on the same issue.

50rayn committed:
Dev: add @unovis/ts packages as dependencies and remove the alias from Webpack

50rayn committed:
Dev: Vite integration

50rayn committed:
React: Vite Integration

50rayn committed:
Dev | React: Dependencies cleanup

rokotyan commented on packages/dev/webpack.config.js:
```diff
@@ -64,7 +64,6 @@ module.exports = {
       '@src': path.resolve(__dirname, './src/'),
 
       // Unovis Core
-      '@unovis/ts': path.resolve(__dirname, '../ts/src/'),
```
Thanks. I'm a bit concerned about these massive changes, and before we merge them, we'll need to thoroughly test how they would affect the dev workflow. Most likely I won't have any bandwidth for this within the next few months, maybe @lee00678 will.

50rayn replied:
Sure thing. It would be perfect if somebody else tested it, just in case I missed something.
The next steps will be:
* PNPM support
* Create a common vite.config ([mergeConfig](https://vite.dev/guide/api-javascript.html#mergeconfig))
* Integrate unit-testing ([vitest monorepo](https://vitest.dev/guide/workspace.html))

lee00678 replied:
I will try to review this in the next 2 weeks.

lee00678 replied:
@50rayn could you please resolve the merge conflicts?

The dev examples seems to be working with the latest updates, but I still see some errors/warnings:

![Screenshot 2025-03-31 at 3 36 03 PM](https://github.com/user-attachments/assets/482c91f0-de2c-4795-92b2-1395d20b70c2)

rokotyan commented:
Thanks @50rayn @lee00678 

I see a couple of problems with merging this:
- When you do `npm run dev`, the development workflow seems to be broken — when you change something in `@unovis/ts` the dev app doesn't show then. If you run build for `@unovis/ts` in the watch mode there are errors and the dev app stops working completely.
- There are import errors in the editor across all packages <img width="959" alt="image" src="https://github.com/user-attachments/assets/02e56754-6273-422d-90a2-7db64a364cbf" />

Maybe we can break it down into two PRs — one for Vite integration and another one with another changes we can keep iterating on,

50rayn commented:
Still, I get some errors when trying to run `npm run generate` from `@unovis/ts`. In the generated components _Graph_, _LeafletMap_ and _LeafletFlowMap_, I get errors after running `nr build:types`. Because of some wrong imports generations, for example for _Graph_:
before:
```
import { Graph, GraphConfigInterface, GraphInputNode, GraphInputLink } from '@unovis/ts'
```

now:
```
import { Graph, GraphConfigInterface } from '@unovis/ts'
import { GraphInputNode, GraphInputLink } from '@/types/graph'
```

Similar to those other components. I try to understand how this magical generation works and why it splitted the imports.

50rayn committed:
TS: Replacing deprecated JSX.Element with React.ReactNode

rokotyan commented:
> Similar to those other components. I try to understand how this magical generation works and why it splitted the imports.

They rely on the paths that come from the TypeScript parser I believe, but it has been a long time since I wrote them, so I'm not 100% confident

50rayn committed:
Dev: Fix Vite migration issues

- Fix tsconfig path aliases for proper ESLint resolution
- Add missing imports (ExampleViewerDurationProps, generateTimeSeries, etc.)
- Fix React key prop warnings in examples
- Fix invalid HTML nesting (<div> inside <p>)
- Add Vite define for UNOVIS_MAP_TILE_SERVER_URL environment variables
- Fix elkjs dynamic import for Vite compatibility
- Exclude elkjs from optimizeDeps to prevent web-worker bundling error
